src/Build/BackEnd/Components/Caching/ConfigCache.cs

```diff
@@ -200,6 +201,20 @@ public void ClearConfigurations()
             }
         }
 
+        /// <summary>
+        /// Gets the smallest configuration id of any configuration
+        /// in this cache.
+        /// </summary>
+        /// <returns>Gets the smallest configuration id of any
+        /// configuration in this cache.</returns>
+        public int GetSmallestConfigId()
+        {
+            lock (_lockObject)
+            {
+                return _configurations.OrderBy(kvp => kvp.Key).FirstOrDefault().Key;
```

Comment:
How often is this called? Linq can be slow, but I think this is called infrequently, so it shouldn't be a problem.

`FirstOrDefault`, if _configurations is empty, will return null, which you then dereference. If it's guaranteed non-null, you can just use `First`. If not, there should be some check in there. (You can consider putting a check in anyway like "if empty, fail.")

Reply:
> How often is this called? Linq can be slow, but I think this is called infrequently, so it shouldn't be a problem.

It would be called once for each project build, at the very end when we serialize the output cache.

> FirstOrDefault, if _configurations is empty, will return null, which you then dereference. If it's guaranteed non-null, you can just use First. If not, there should be some check in there. (You can consider putting a check in anyway like "if empty, fail.")

It should always be non-null, but I'll put the check in there for safety.